quanto/tensor/core.py

```diff
@@ -36,6 +37,94 @@ def axis_to_dim(t, axis):
     return dim
 
 
+
+def pack_weights(intweights: torch.Tensor, bitsdtype: qbitsdtype) -> torch.Tensor:
+    """
+    Pack int4 / int2 weights in a int32 tensor
+    What packing means? Assume we have 8 values that are in 4bit but encoded in 8bit 
+    (because torch and numpy does not have native support for 4-bit datatypes)
+
+    > 0000 0101 | 0000 1110 | 0000 1010 | 0000 1111 | 0000 0101 | 0000 1110 | 0000 1010 | 0000 1111 
+
+    We can pack them in a single 32-bit uint tensor
+
+    > 0101 1110 1010 1111 0101 1110 1010 1111
+
+    Therefore instead of saving a 8 tensors in 8-bit precision we save a single tensor of 32-bit precision saving 32-bits in total.
+    However pytorch does not come with native support of `uint32` datatype, therefore we need to first
+    create a uint32 tensor in numpy and convert it to a `torch.int32` tensor.
+
+    Args:
+        intweights (`torch.Tensor`):
+            The un-packed tensor in 8-bit precision
```

Comment:
maybe add an assert that the tensor should be of dim 2

Reply:
done!